public/app/features/dashboard-scene/scene/VizPanelHeaderActions.tsx

```diff
@@ -0,0 +1,136 @@
+import { Unsubscribable } from 'rxjs';
+
+import {
+  GroupByVariable,
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneQueryRunner,
+  VizPanel,
+} from '@grafana/scenes';
+import { DataSourceRef } from '@grafana/schema';
+
+import { verifyDrilldownApplicability } from '../utils/drilldownUtils';
+
+import { PanelGroupByAction } from './panel-actions/PanelGroupByAction/PanelGroupByAction';
+
+export interface VizPanelHeaderActionsState extends SceneObjectState {
+  hideGroupByAction?: boolean;
```

Comment:
Nit: I'd inverse this to be `useGroupByAction / showGroupByAction`. Would be just dependent on the toggle, rather than more complex to process negation of the toggle.